cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/CDAPLogAppender.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.framework;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+import ch.qos.logback.core.LogbackException;
+import ch.qos.logback.core.spi.FilterReply;
+import ch.qos.logback.core.status.WarnStatus;
+import co.cask.cdap.logging.serialize.LogSchema;
+import co.cask.cdap.logging.write.FileMetaDataManager;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Flushable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Log Appender implementation for CDAP Log framework
+ * TODO : Refactor package CDAP-8196
+ */
+public class CDAPLogAppender extends AppenderBase<ILoggingEvent> implements Flushable {
+  private static final Logger LOG = LoggerFactory.getLogger(CDAPLogAppender.class);
+  public static final String TAG_NAMESPACE_ID = ".namespaceId";
+  public static final String TAG_APPLICATION_ID = ".applicationId";
+  public static final String TAG_FLOW_ID = ".flowId";
+  public static final String TAG_COMPONENT_ID = ".componentId";
+  public static final String TAG_MAP_REDUCE_JOB_ID = ".mapReduceId";
+  public static final String TAG_SPARK_JOB_ID = ".sparkId";
+  public static final String TAG_USER_SERVICE_ID = ".userserviceid";
+  public static final String TAG_WORKER_ID = ".workerid";
+  public static final String TAG_WORKFLOW_ID = ".workflowId";
+
+  private LogFileManager logFileManager;
+
+  @Inject
+  private FileMetaDataManager fileMetaDataManager;
+  @Inject
+  private LocationFactory locationFactory;
+
+  private int syncIntervalBytes;
+  private long maxFileLifetimeMs;
+
+
+  /**
+   * TODO: start a separate cleanup thread to remove files that has passed the TTL
```

Comment:
Should we create a jira for it and mark it in `4.1.0`? If there is any exception while doing log clean up, the CDAP appender might fail right?